UnityProject/Assets/Scripts/Items/Weapons/Melee/MeleeEffect.cs

```diff
@@ -9,30 +10,40 @@ namespace Weapons
 	/// </summary>
 	public class MeleeEffect : MonoBehaviour, ICheckedInteractable<HandApply>, IServerSpawn
 	{
+		/// <summary>
+		/// Sounds to play when striking someone
+		/// </summary>
+		[SerializeField] private AddressableAudioSource useSound = null;
 
-		[HideInInspector]
+		[Tooltip("How long you have to wait before triggering this weapons effect again.")]
+		[SerializeField]
+		[Space(10)]
+		private int cooldown = 5;
+
+		[Space(10)]
 		[Tooltip("Does this weapon stun players on hit?")]
 		public bool canStun = false;
 
-		/// <summary>
-		/// How long to stun for (in seconds)
-		/// </summary>
-		[HideInInspector]
+		[ShowIf("canStun")]
+		[Tooltip("How long to stun for (in seconds)")]
 		public float stunTime = 0;
 
-		[HideInInspector]
+		[Space(10)]
 		[Tooltip("Does this weapon teleport players on hit?")]
 		public bool canTeleport = false;
 
-		[HideInInspector]
+		[ShowIf("canTeleport")]
 		public bool avoidSpace;
-		[HideInInspector]
+		[ShowIf("canTeleport")]
 		public bool avoidImpassable = true;
 
-		[HideInInspector]
+		[ShowIf("canTeleport")]
```

Comment:
change all the ``ShowIf`` to use ``nameof()`` instead of a string so if the field name changes in the future, the code doesn't silently stop working

Reply:
in this case it would be ``ShowIf(nameof(canTeleport))``